src/jobs/anonymizeBeneficiaries.ts

```diff
@@ -3,63 +3,72 @@ import { Types } from 'mongoose';
-    const aidRecord = allAids[i];
+  // Get all the family records
+  const allFamilies = await Record.find({
+    resource: FAMILY_RESOURCE_ID,
+  });
 
-    // Get Family record
-    const familyRecord = await Record.findOne({
-      _id: aidRecord?.data?.owner_resource,
+  // For each family record, check if exists
+  // an aid record in the last 18 months
+  for (const family of allFamilies) {
+    const aidGivenToFamily = await Record.exists({
+      resource: AID_RESOURCE_ID,
+      createdAt: {
+        $gt: new Date(Date.now() - 18 * 30 * 24 * 60 * 60 * 1000),
+      }, // 18 months ago
+      'data.owner_resource': family._id.toString(),
     });
 
-    // Find all members of the family
-    const members = await Record.find({
-      _id: { $in: familyRecord?.data?.members },
-    });
+    // If no aid was given to the family in the last 18 months
+    if (!aidGivenToFamily) {
+      // Find all members of the family
+      const members = await Record.find({
+        _id: { $in: family?.data?.members },
+      });
 
-    // Anonymize all members of that family
-    for (let j = 0; j < members.length; j++) {
-      const member = members[j];
+      // Anonymize all the members
+      members.forEach((member) => {
+        if (!member.data) {
+          return;
+        }
+        // Anonymize the member
+        member._createdBy = new User({
+          name: 'ANONYMOUS',
+          username: `${member._id.toString()}@oort-anonymous.com`,
+        });
 
-      // Anonymize the member
-      member._createdBy = new User({
-        name: 'ANONYMOUS',
-        username: `${Math.random()
-          .toString(36)
-          .substring(2, 15)}@anonymus-oort.com`,
-      });
-      member.data.location = 'ANONYMOUS';
-      member.data.surname = 'ANONYMOUS';
-      member.data.firstname = 'ANONYMOUS';
-      member.data.phone = 'ANONYMOUS';
-      member.data.nom_employes = 'ANONYMOUS';
-      member.data.gender = 'ANONYMOUS';
-      member.data.birthdate = 'ANONYMOUS';
-      member.data.prenom_employes = 'ANONYMOUS';
-      member.data.nom_prenom_employes = 'ANONYMOUS';
-      member.data.tel_staff = 'ANONYMOUS';
-      member.data.email_staff = 'ANONYMOUS';
-      member.data.birthdate_employes = 'ANONYMOUS';
-      member._lastUpdatedBy = new User({
-        name: 'ANONYMOUS',
-        username: `${Math.random()
-          .toString(36)
-          .substring(2, 15)}@anonymus-oort.com`,
+        member.data = {
+          ...member.data,
+          location: 'ANONYMOUS',
+          surname: 'ANONYMOUS',
+          firstname: 'ANONYMOUS',
+          phone: 'ANONYMOUS',
+          nom_employes: 'ANONYMOUS',
+          gender: 'ANONYMOUS',
+          birthdate: 'ANONYMOUS',
+          prenom_employes: 'ANONYMOUS',
+          nom_prenom_employes: 'ANONYMOUS',
+          tel_staff: 'ANONYMOUS',
+          email_staff: 'ANONYMOUS',
+          birthdate_employes: 'ANONYMOUS',
+          file_gdpr_staff: [],
+        };
```

Comment:
When anonymizing the data, we don't need to remove fields that are not traceable back to the beneficiary. e.g: their location and gender. 

Also, make sure to only save valid data. For the birthdate for example, it should always be a valid date, check the other anonymization function to see how it should be dealt with. The email should also be a valid email, you can use the same logic of `[id]@oort-anonymous.com`. To make sure that all fields have valid data, you can create a record and see how the data is stored in the db.

And finally, it looks like there are fields here that are not part of the form (and it's possibly missing some that should be there, but I didn't check)